src/admin/components/Alerts/DelayedJobsProgressAlert.tsx

```diff
@@ -12,47 +11,16 @@ type DelayedJobsProgressAlertProps = {
 };
 
 const DelayedJobsProgressAlert: FC<DelayedJobsProgressAlertProps> = ({ show, title, setIsLoadingDelayedJob }) => {
-  const [delayedJobProcessing, setDelayedJobProcessing] = useState<number>(0);
-  const [delayedJobTotal, setDalayedJobTotal] = useState<number>(0);
-  const [progressMessage, setProgressMessage] = useState<string>("Running 0 out of 0 polygons (0%)");
-
-  const store = useStore<AppStore>();
-  useEffect(() => {
-    let intervalId: any;
-    if (show) {
-      intervalId = setInterval(() => {
-        const { total_content, processed_content, progress_message } = store.getState().api;
-        setDalayedJobTotal(total_content);
-        setDelayedJobProcessing(processed_content);
-        if (progress_message != "") {
-          setProgressMessage(progress_message);
-        }
-      }, 1000);
-    }
-
-    return () => {
-      if (intervalId) {
-        setDelayedJobProcessing(0);
-        setDalayedJobTotal(0);
-        setProgressMessage("Running 0 out of 0 polygons (0%)");
-        clearInterval(intervalId);
-      }
-    };
-  }, [show]);
+  const { totalContent, processedContent, progressMessage } = useJobProgress();
 
   const abortDelayedJob = () => {
-    ApiSlice.abortDelayedJob(true);
-    ApiSlice.addTotalContent(0);
-    ApiSlice.addProgressContent(0);
-    ApiSlice.addProgressMessage("Running 0 out of 0 polygons (0%)");
```

Comment:
This was the other pattern I wanted to call out. Calling several redux store updates in a row is an anti-pattern. Much better to have a single action dispatcher that issues all the changes in a single event. Because of the careful processing of caching selectors, this kind of the thrashing, while not a huge deal, is a performance bottleneck that we'd like to avoid.